tools/generator/filters/c.py

```diff
@@ -10,5 +12,17 @@ def c_type(value: str) -> str:
     return value
 
 
-c_args = functools.partial(generic_args, type_mapper=c_type)
-c_prototype = functools.partial(generic_prototype, arg_mapper=c_args)
+@register_filter
+def c_args(value):
+    return generic_args(value, type_mapper=c_type)
+
+
+@register_filter
+def c_prototype(value):
+    return generic_prototype(value, arg_mapper=c_args)
+
+
+@register_filter
+@jinja2.environmentfilter
+def c_comment(env, value, indent: int = 0):
+    return generic_comment(env, value=value, start='// ', indent=indent)
```

Comment:
This whole section is the boilerplate I'm talking about, that will basically need to be copypasted across all languages.

Yes, I know how to make it more generic, but it comes at the cost of readability. We know how unreadable it can become in the case of the previous generators when too much magic is involved.

Reply:
I don't mind verbosity when it's making stuff more maintainable in the long run.

Reply:
Alternative "magic" option to replace all these lines:

```
c_args = register_partial_filter('c_args', generic_args, type_mapper=c_type)
c_args = register_partial_filter('c_prototype', generic_args, type_mapper=c_args)
c_comment = register_partial_filter('c_comment', generic_comment, start='// ')
```